src/resources/ResourceQuotas/support.ts

```diff
@@ -0,0 +1,105 @@
+import {
+  bytesToHumanReadable,
+  cpusToHumanReadable,
+  getBytes,
+  getCpus,
+} from 'resources/Namespaces/ResourcesUsage';
+import { ResourceQuotaProps } from './ResourceQuotaDetails';
+import { roundTwoDecimals } from 'shared/utils/helpers';
+import { calculateMetrics } from 'resources/Pods/podQueries';
+import { UsageMetrics } from 'resources/Pods/types';
+import { isEmpty } from 'lodash';
+
+// It takes resource, usage and limits data
+// and maps it to data suitable for charts to make it easier to iterate and render.
+export const mapLimitsAndUsageToChartsData = (
+  resource?: ResourceQuotaProps,
+  podsMetrics?: UsageMetrics[],
+) => {
+  const totalLimitsCpu = getCpus(resource?.status?.hard?.['limits.cpu']);
+  const totalLimitsUsageCpu = getCpus(resource?.status?.used?.['limits.cpu']);
+  const totalLimits = getBytes(resource?.status?.hard?.['limits.memory']);
+  const totalLimitsUsage = getBytes(resource?.status?.used?.['limits.memory']);
+  const totalRequestsCpu = getCpus(resource?.status?.hard?.['requests.cpu']);
+  const totalRequestsUsageCpu = getCpus(
+    resource?.status?.used?.['requests.cpu'],
+  );
+  const totalRequests = getBytes(
+    resource?.status?.hard?.['requests.memory'] ??
+      resource?.status?.hard?.memory,
+  );
+  const totalRequestsUsage = getBytes(
+    resource?.status?.used?.['requests.memory'] ??
+      resource?.status?.used?.memory,
+  );
+  const { cpu, memory } = calculateMetrics(podsMetrics);
+
+  return [
```

Comment:
Can we check length of `podsMetrics` at the beginning and return []?
There is not reason to do calculation when this condition is so important.